### PR TITLE
STYLE: Added explicit exceptions 

### LINKED ISSUE #22877: Replace bare excepts by explicit excepts in pandas
(superseeds #18419, same task for different files: #22872, #22873, #22874, #22875)

In several parts of the code, we have bare excepts in the form of:
```python
try:
    my_value = my_dict[my_key]
except:  # we are capturing all the exceptions, when we want to capture only KeyError
    my_value = None
```

This is a bad practice, and we want to avoid having this in the code. What we want instead is:
```python
try:
    my_value = my_dict[my_key]
except KeyError:
    my_value = None
```
Of course in every case, the exception can be different (not always `KeyError`), and some research is needed to see which is the right exception (or exceptions, it can be a tuple) that needs to be captured. In some cases, every exception needs to be captured and we'll use `except Exception:`, but this should be avoided unless we really need to capture all exceptions.

This issue is to fix all the bare excepts in testing files `pandas/tests/`. At the moment they are (note that the list can change as code evolves):
```
pandas/compat/pickle_compat.py:36:13: E722 do not use bare except'
pandas/compat/pickle_compat.py:47:13: E722 do not use bare except'
pandas/compat/pickle_compat.py:185:1: E722 do not use bare except'
pandas/compat/pickle_compat.py:213:5: E722 do not use bare except'
pandas/tseries/holiday.py:297:5: E722 do not use bare except'
pandas/tseries/holiday.py:429:9: E722 do not use bare except'
pandas/tseries/holiday.py:438:9: E722 do not use bare except'
pandas/util/_print_versions.py:24:9: E722 do not use bare except'
pandas/util/_print_versions.py:53:5: E722 do not use bare except'
pandas/util/_print_versions.py:111:9: E722 do not use bare except'
pandas/util/_validators.py:62:9: E722 do not use bare except'
```
An updated list can be obtained by running: `flake8 --select=E722 --config=none --exclude=pandas/tests,pandas/io,pandas/core pandas`

### PR DESCRIPTION
**Pull request by NeuralFlux**
- [x] xref  #22877
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Fixed the following bare excepts:

> pandas/compat/pickle_compat.py:36:13: E722 do not use bare except'
> pandas/compat/pickle_compat.py:47:13: E722 do not use bare except'
> pandas/compat/pickle_compat.py:185:1: E722 do not use bare except'
> pandas/compat/pickle_compat.py:213:5: E722 do not use bare except'
> pandas/tseries/holiday.py:297:5: E722 do not use bare except'
> pandas/tseries/holiday.py:429:9: E722 do not use bare except'
> pandas/tseries/holiday.py:438:9: E722 do not use bare except'
> pandas/util/_print_versions.py:24:9: E722 do not use bare except'
> pandas/util/_print_versions.py:53:5: E722 do not use bare except'
> pandas/util/_print_versions.py:111:9: E722 do not use bare except'
> pandas/util/_validators.py:62:9: E722 do not use bare except'

### CODE REVIEW

File: pandas/util/_print_versions.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def show_versions(as_json=False):
                 mod = importlib.import_module(modname)
             ver = ver_f(mod)
             deps_blob.append((modname, ver))
-        except:
+        except ModuleNotFoundError:
```
Comment (datapythonista) [inline pandas/util/_print_versions.py]: Great PR. `ModuleNotFoundError` is Python 3 only, and unfortunately we need to support Python 2 few more months. Can you find a Python 2 compatible way for this please?

**Review by datapythonista — APPROVED** [review]
Thanks for the PR @NeuralFlux, looks great. I think it's ready to be merged when the checks are green. @jreback

**Comment by NeuralFlux** [comment]
Thank you so much @datapythonista ! Looking forward for those green checks!

**Comment by jreback** [comment]
thanks @NeuralFlux

**Comment by NeuralFlux** [comment]
> 
> 
> thanks @NeuralFlux

My pleasure to help! Thank you for the good-first-issue! @jreback and @datapythonista

**Pull request merged**